src/meepgeom.cpp

```diff
@@ -1768,15 +1770,21 @@ void fragment_stats::compute_stats() {
     geometric_object *go = &geom.items[i];
     double overlap = box_overlap_with_object(box, *go, tol, maxeval);
 
-    // If the object doesn't overlap the entire box, that implies that
-    // an object interface intercepts the box, which means we treat
-    // the entire box as anisotropic.
     bool anisotropic_pixels_already_added = false;
-    if (overlap != 1.0) {
-      anisotropic_pixels_already_added = true;
-      num_anisotropic_eps_pixels += num_pixels_in_box;
-      if (mu_not_1(go->material)) {
-        num_anisotropic_mu_pixels += num_pixels_in_box;
+    if (eps_averaging) {
+      // If the object doesn't overlap the entire box, that implies that
+      // an object interface intercepts the box, which means we treat
+      // the entire box as anisotropic. This method could give some false
+      // positives if there is another object with the same material behind
+      // the current object, but in practice it is probably reasonable to
+      // assume that there is a material interface somwhere in the box so
+      // we won't worry about fancier edge-detection methods for now.
+      if (overlap != 1.0) {
+        anisotropic_pixels_already_added = true;
+        num_anisotropic_eps_pixels += num_pixels_in_box;
+        if (mu_not_1(go->material)) {
```

Comment:
This could give false negatives, since `go` could be something like vacuum, and it could be the material behind with mu.

But mu is not that common anyway so I'm inclined not to worry about it for now.